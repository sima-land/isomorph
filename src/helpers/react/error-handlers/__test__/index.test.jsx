import React from 'react';
import { mount, shallow } from 'enzyme';
import { ErrorBoundary, SafeSuspense } from '../index';

describe('ErrorBoundary', () => {
  const error = new Error('Test error');

  const FailedComponent = () => {
    throw error;
  };
  it('should render children component', () => {
    const wrapper = mount(
      <ErrorBoundary fallback={null}>
        <div>Normal component</div>
      </ErrorBoundary>
    );

    expect(wrapper.text()).toContain('Normal component');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render fallback component', () => {
    const wrapper = mount(
      <ErrorBoundary fallback={<div>Fallback</div>}>
        <>
          <div>Normal component</div>
          <FailedComponent />
        </>
      </ErrorBoundary>
    );

    expect(wrapper.text()).toContain('Fallback');
    expect(wrapper).toMatchSnapshot();
  });

  it('should call captureException', () => {
    const handlerException = jest.fn();

    const wrapper = mount(
      <ErrorBoundary captureException={handlerException} fallback={null}>
          <>
            <div>Normal component</div>
            <FailedComponent />
          </>
      </ErrorBoundary>
    );

    expect(handlerException).toBeCalledWith(error, { componentStack: expect.any(String) });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('SafeSuspense', () => {
  it('should render with props', () => {
    const wrapper = shallow(
      <SafeSuspense
        fallback={<div>Fallback</div>}
        captureException={jest.fn()}
      >
        <div>Normal component</div>
      </SafeSuspense>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
