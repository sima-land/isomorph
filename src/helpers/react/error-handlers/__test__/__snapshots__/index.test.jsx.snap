// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should call captureException 1`] = `
<ErrorBoundary
  captureException={
    [MockFunction] {
      "calls": Array [
        Array [
          [Error: Test error],
          Object {
            "componentStack": "
    in FailedComponent
    in ErrorBoundary (created by WrapperComponent)
    in WrapperComponent",
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  fallback={null}
/>
`;

exports[`ErrorBoundary should render children component 1`] = `
<ErrorBoundary
  fallback={null}
>
  <div>
    Normal component
  </div>
</ErrorBoundary>
`;

exports[`ErrorBoundary should render fallback component 1`] = `
<ErrorBoundary
  fallback={
    <div>
      Fallback
    </div>
  }
>
  <div>
    Fallback
  </div>
</ErrorBoundary>
`;

exports[`SafeSuspense should render with props 1`] = `
<Suspense
  fallback={
    <div>
      Fallback
    </div>
  }
>
  <ErrorBoundary
    captureException={[MockFunction]}
    fallback={
      <div>
        Fallback
      </div>
    }
  >
    <div>
      Normal component
    </div>
  </ErrorBoundary>
</Suspense>
`;
